HB/instance.elpi

```diff
@@ -180,9 +170,12 @@ declare-canonical-instances-from-factory T F :-
   % priority must be given to canonical mixins
   % as they are the ones which guarantee forgetful inheritance
   % hence we add these clauses last.
-  under-mixin-src-from-factory.do! T F [
-    under-canonical-mixins-of.do! T [
-      add-all-mixins T GRF {list-w-params_list {factory-provides GRF}} tt _,
+  synthesis.under-mixin-src-from-factory.do! T F [
+    synthesis.under-local-canonical-mixins-of.do! T [
+      list-w-params_list {factory-provides GRF} ML,
+      std.map ML (synthesis.assert!-infer-mixin T) MI,
+      std.zip ML MI MLI, 
+      add-all-mixins T GRF MLI tt _,
```

Comment:
@CohenCyril I need helpt to understand your comment. Also, in both the old and new code the mixins provided by GRF are defined by we don't load the clauses which make them the preferred way to build these mixins ouf of these definitions (add-all-mixins returns mixin-src clauses we drop)